lib/abci/handlers/query/identityQueryHandlerFactory.js

```diff
@@ -9,26 +9,44 @@ const NotFoundAbciError = require('../../errors/NotFoundAbciError');
 /**
  *
  * @param {IdentityStoreRepository} identityRepository
+ * @param {RootTree} rootTree
+ * @param {IdentitiesStoreRootTreeLeaf} identitiesStoreRootTreeLeaf
  * @return {identityQueryHandler}
  */
-function identityQueryHandlerFactory(identityRepository) {
+function identityQueryHandlerFactory(
+  identityRepository,
+  rootTree,
+  identitiesStoreRootTreeLeaf,
+) {
   /**
    * @typedef identityQueryHandler
    * @param {Object} params
    * @param {Object} options
    * @param {Buffer} options.id
+   * @param {Object} request
+   * @param {boolean} [request.prove]
    * @return {Promise<ResponseQuery>}
    */
-  async function identityQueryHandler(params, { id }) {
+  async function identityQueryHandler(params, { id }, request) {
+    const includeProof = request.prove === 'true';
+
     const identity = await identityRepository.fetch(id);
 
     if (!identity) {
       throw new NotFoundAbciError('Identity not found');
     }
 
-    return new ResponseQuery({
-      value: identity.toBuffer(),
-    });
+    const identityBuffer = identity.toBuffer();
+
+    const result = {
+      value: identityBuffer,
+    };
+
+    if (includeProof) {
+      result.proof = rootTree.getFullProof(identitiesStoreRootTreeLeaf, [identityBuffer]);
```

Comment:
Shouldn't it be Identity ID?

Reply:
Also, it seems Tendermint expects a different response https://docs.tendermint.com/master/spec/abci/abci.html#proof